src/Symfony/Component/Messenger/Envelope.php

```diff
@@ -17,8 +17,10 @@
  * A message wrapped in an envelope with stamps (configurations, markers, ...).
  *
  * @author Maxime Steinhausser <maxime.steinhausser@gmail.com>
+ *
+ * @final
  */
-final class Envelope
+class Envelope
```

Comment:
Why is the reason do remove PHP final here? I understand that the `@final` is not enforced, probably to allow people to override the class, but then what's the point?

Reply:
The tests require the change for mocks.

Reply:
(In order to mock the new return values.)

Reply:
You could consider to introduce an interface or to just create the Envelope object in tests (instead of mocking it). I don't see it mocked anywhere by the way.

Reply:
Data object -> no mocks. In any case, removing `final` "just" for the tests sounds like a terrible idea.

Reply:
I agree with @fabpot that Envelope does not need to be mocked and thus could remain final.

Reply:
reverted, the class is now really final again